src/js/stores/MesosStateStore.js

```diff
@@ -289,8 +289,11 @@ class MesosStateStore extends GetSetBaseStore {
     // the scheduler tasks or a list of Marathon application tasks.
     return frameworks.reduce(function(serviceTasks, framework) {
       const { tasks = [], completed_tasks = {}, name } = framework;
-      // Include tasks from framework match, if service is a Framework
-      if (service instanceof Framework && name === serviceName) {
+
+      if (
```

Comment:
Not blocking, but I'd actually get the framework name only once to reduce the number of call. We might also be able to do the `Framework` type check only once.